dashboard-server/queries/responses/response.test.js

```diff
@@ -57,33 +65,84 @@ describe('Response queries', () => {
-       ,('${survey.id}', '101', '${survey.id}', '101', 100006, '124', 'ref', 10, 'text', '{ "text": "middle" }', '6789', current_date::timestamptz + interval '12 hour')`;
+        ('${survey.id}', '101', '${
+        survey.id
+      }', '101', 100001, '124', 'ref', 10, 'text', '{ "text": "last" }', '6789', current_date::timestamptz + interval '14 hour')
+       ,('${survey2.id}', '202', '${
+        survey2.id
+      }', '202', 100003, '123', 'ref', 10, 'text', '{ "text": "last" }', '6789', (date '2019-04-18')::timestamptz + interval '12 hour')
+       ,('${survey.id}', '101', '${
+        survey.id
+      }', '101', 100004, '124', 'ref', 10, 'text', '{ "text": "first" }', '6789', current_date::timestamptz + interval '10 hour')
+       ,('${survey2.id}', '202', '${
+        survey2.id
+      }', '202', 100005, '123', 'ref', 10, 'text', '{ "text": "first" }', '6789', (date '2019-04-18')::timestamptz + interval '8 hour')
+       ,('${survey2.id}', '202', '${
+        survey2.id
+      }', '202', 100003, '125', 'ref', 10, 'text', '{ "text": "last" }', '6789', (date '2019-04-18')::timestamptz + interval '12 hour')
+       ,('${survey.id}', '101', '${
+        survey.id
+      }', '101', 100004, '125', 'ref', 10, 'text', '{ "text": "first" }', '6789', (date '2019-04-18')::timestamptz + interval '10 hour')
+       ,('${survey2.id}', '202', '${
+        survey2.id
+      }', '202', 100005, '125', 'ref', 10, 'text', '{ "text": "first" }', '6789', (date '2019-04-18')::timestamptz + interval '8 hour')
+       ,('${survey.id}', '101', '${
+        survey.id
+      }', '101', 100006, '124', 'ref', 10, 'text', '{ "text": "middle" }', '6789', current_date::timestamptz + interval '12 hour')`;
 
       await vlabPool.query(MOCK_QUERY);
-      const responses = await Response.all();
+      const responses = await Response.firstAndLast();
 
       responses[0].first_response.should.equal('{ "text": "first" }');
       responses[0].last_response.should.equal('{ "text": "last" }');
-      responses[0].surveyid.should.equal(
-        survey2.id,
-      );
+      responses[0].surveyid.should.equal(survey2.id);
       responses[1].first_response.should.equal('{ "text": "first" }');
       responses[1].last_response.should.equal('{ "text": "last" }');
-      responses[1].surveyid.should.equal(
-        survey.id,
+      responses[1].surveyid.should.equal(survey.id);
+    });
+  });
+
+  describe('.all()', () => {
+    it('should return all responses for a survey created by a user', async () => {
+      const responses = await Response.all({
+        email: 'test2@vlab.com',
+        survey: 'Survey',
+      });
+
+      const responses2 = await Response.all({
+        email: 'test2@vlab.com',
+        survey: 'Survey123',
+      });
+
+      const responses3 = await Response.all({
+        email: 'test3@vlab.com',
+        survey: 'Survey',
+      });
+      responses.length.should.equal(8);
```

Comment:
Great job with other users, but also would be good to have some responses that aren't returned!